### DISCUSSION TITLE
Search Bar

### DISCUSSION THREAD

**Discussion by mic-t** [DirectoryLister/DirectoryLister#730] [Help & Support]
When searching, no results are returned. Seems to be the same behavior with a wildcard search or when searching for a specific file name.  the icon in the browser tab spins until interrupted, no error information reported to the screen. 
.env  looks like this:

APP_DEBUG=true
APP_LANGUAGE=en

DISPLAY_READMES=true
READMES_FIRST=false
ZIP_DOWNLOADS=false

GOOGLE_ANALYTICS_ID=false

MATOMO_ANALYTICS_URL=false
MATOMO_ANALYTICS_ID=false

SORT_ORDER=type
REVERSE_SORT=false

--
Any ideas will be appreciated :-)

**Comment by mic-t** [comment]
Found this in the apache2 log:

PHP Fatal error:  Maximum execution time of 300 seconds exceeded in /data/text/app/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 78, referer: http://myserver.unk.orgorg/textfiles/
*
PHP Fatal error:  Maximum execution time of 300 seconds exceeded in 
/data/text/app/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 78, referer: http://myserver.unk,.org/textfiles/?dir=20150930
**
Same issue in the install root dir and a subdir

**Reply by PHLAK** [reply]
How many files are in your listing total (including files in sub-folders)? You can use `find` + `wc` to find this out.

```
find /path/to/your/listing -type f | wc -l
```

**Reply by mic-t** [reply]
There are 2,955,931 text files in about 1250 sub dirs
--
not sure if it is interesting, but it is an Alias. 
All text files, half are 1 line, the others average size ranges from 1k -5k

**Reply by PHLAK** [reply]
This is definitely the core of issue. Under the hood the search functionality builds an array of every file in the listing. I was aware of the potential for this to cause memory issues with extraordinarily large listings but didn't know exactly how large. I will look into optimizing this at some point but can't promise that any time soon.

In the mean time you may be able to alleviate this by increasing the memory limit or, if possible, splitting up your listing into completely separate Directory Lister instances each with a manageable subset of files.

**Reply by visualblind** [reply]
> This is definitely the core of issue. Under the hood the search functionality builds an array of every file in the listing. I was aware of the potential for this to cause memory issues with extraordinarily large listings but didn't know exactly how large. I will look into optimizing this at some point but can't promise that any time soon.
> 
> In the mean time you may be able to alleviate this by increasing the memory limit or, if possible, splitting up your listing into completely separate Directory Lister instances each with a manageable subset of files.

This probably should have been an easy foresight due to not knowing end users datastore sizes. I'm sharing about 30TB which is many more files than an unindexed search function should even think about tackling. To even try is almost funny.
If there is a unofficial way to disable the search function that would be great. Otherwise I'll probably need to hide it in CSS because runaway searches are causing long-lived php processes chewing up valuable cpu time via mergerfs.
I have tried configuring `request_terminate_timeout` in the php pool, however that also breaks directory zip downloading.

**\*Update\*** If you want to hide the Search form from view, edit `app/assets/app.css` and add `form{visibility:hidden}`. I added it to the end of line 2 only because I thought that was the end of the file (it wasn't :laughing:), like so:
```
@import url(https://fonts.googleapis.com/css2?family=Source+Code+Pro&family=Work+Sans&display=swap);
/*! tailwindcss v3.4.10 | MIT License | https://tailwindcss.com*/*,:after,:before{border:0 solid #e2e8f0;box-sizing:border-box}:after,:before{--tw-content:""}:host,html{-webkit-text-size-adjust:100%;font-feature-settings:normal;-webkit-tap-highlight-color:transparent;font-family:Work Sans,ui-sans-serif,system-ui,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;font-variation-settings:normal;line-height:1.5;-moz-tab-size:4;-o-tab-size:4;tab-size:4}body{line-height:inherit;margin:0}hr{border-top-width:1px;color:inherit;height:0}abbr:where([title]){-webkit-text-decoration:underline dotted;text-decoration:underline dotted}h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit}a{color:inherit;text-decoration:inherit}b,strong{font-weight:bolder}code,kbd,pre,samp{font-feature-settings:normal;font-family:Source Code Pro,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;font-size:1em;font-variation-settings:normal}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}table{border-collapse:collapse;border-color:inherit;text-indent:0}button,input,optgroup,select,textarea{font-feature-settings:inherit;color:inherit;font-family:inherit;font-size:100%;font-variation-settings:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;margin:0;padding:0}button,select{text-transform:none}button,input:where([type=button]),input:where([type=reset]),input:where([type=submit]){-webkit-appearance:button;background-color:transparent;background-image:none}:-moz-focusring{outline:auto}:-moz-ui-invalid{box-shadow:none}progress{vertical-align:baseline}::-webkit-inner-spin-button,::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}summary{display:list-item}blockquote,dd,dl,figure,h1,h2,h3,h4,h5,h6,hr,p,pre{margin:0}fieldset{margin:0}fieldset,legend{padding:0}menu,ol,ul{list-style:none;margin:0;padding:0}dialog{padding:0}textarea{resize:vertical}input::-moz-placeholder,textarea::-moz-placeholder{color:#94a3b8;opacity:1}input::placeholder,textarea::placeholder{color:#94a3b8;opacity:1}[role=button],button{cursor:pointer}:disabled{cursor:default}audio,canvas,embed,iframe,img,object,svg,video{display:block;vertical-align:middle}img,video{height:auto;max-width:100%}[hidden]{display:none}*,:after,:before{--tw-border-spacing-x:0;--tw-border-spacing-y:0;--tw-translate-x:0;--tw-translate-y:0;--tw-rotate:0;--tw-skew-x:0;--tw-skew-y:0;--tw-scale-x:1;--tw-scale-y:1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness:proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:rgba(14,165,233,.5);--tw-ring-offset-shadow:0 0 #0000;--tw-ring-shadow:0 0 #0000;--tw-shadow:0 0 #0000;--tw-shadow-colored:0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }::backdrop{--tw-border-spacing-x:0;--tw-border-spacing-y:0;--tw-translate-x:0;--tw-translate-y:0;--tw-rotate:0;--tw-skew-x:0;--tw-skew-y:0;--tw-scale-x:1;--tw-scale-y:1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness:proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:rgba(14,165,233,.5);--tw-ring-offset-shadow:0 0 #0000;--tw-ring-shadow:0 0 #0000;--tw-shadow:0 0 #0000;--tw-shadow-colored:0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }.container{width:100%}@media (min-width:640px){.container{max-width:640px}}@media (min-width:768px){.container{max-width:768px}}@media (min-width:1024px){.container{max-width:1024px}}@media (min-width:1280px){.container{max-width:1280px}}@media (min-width:1536px){.container{max-width:1536px}}form{visibility:hidden}
```

And Voilà!

![Screenshot_2024-10-08_06-23-18](https://github.com/user-attachments/assets/0904a0c0-6f16-485f-802d-a3b8d1970861)
